packages/core/core/src/services/document-service/webhooks.ts

```diff
@@ -0,0 +1,82 @@
+import { curry } from 'lodash/fp';
+
+import { UID, Schema, Utils, Core } from '@strapi/types';
+import { sanitize } from '@strapi/utils';
+
+import { getDeepPopulate } from './utils/populate';
+
+const ALLOWED_WEBHOOK_EVENTS = {
+  ENTRY_CREATE: 'entry.create',
+  ENTRY_UPDATE: 'entry.update',
+  ENTRY_DELETE: 'entry.delete',
+  ENTRY_PUBLISH: 'entry.publish',
+  ENTRY_UNPUBLISH: 'entry.unpublish',
+  ENTRY_DRAFT_DISCARD: 'entry.draft-discard',
+};
+
+type WebhookEvent = Utils.Object.Values<typeof ALLOWED_WEBHOOK_EVENTS>;
+
+const sanitizeEntry = async (
+  model: Schema.ContentType<any> | Schema.Component<any>,
+  entry: any
```

Comment:
Isn't there a type like Modules.Documents.AnyDocument? Is that not what this would be here and below for emitWebHoook's entry arg?

Reply:
will check! good point

Reply:
spot on! used tha xact same type :)